metrics/src/beaconchain.rs

```diff
@@ -88,29 +91,13 @@ pub struct ProcessMetrics {
 
 impl ProcessMetrics {
     pub fn get() -> Self {
-        let mut cpu_process_seconds_total = 0;
-        let mut memory_process_bytes = 0;
-
-        match Process::current() {
-            Ok(process) => {
-                match process.cpu_times() {
-                    Ok(cpu_times) => {
-                        cpu_process_seconds_total = cpu_times.busy().as_secs()
-                            + cpu_times.children_system().as_secs()
-                            + cpu_times.children_system().as_secs();
-                    }
-                    Err(error) => warn!("unable to get current process CPU usage: {error:?}"),
-                }
-
-                match process.memory_info() {
-                    Ok(mem_info) => {
-                        memory_process_bytes = mem_info.rss();
-                    }
-                    Err(error) => warn!("unable to get process memory usage: {error:?}"),
-                }
-            }
-            Err(error) => warn!("unable to get current process: {error:?}"),
-        }
+        let Ok(crate::metric_sys::ProcessCpuMetric {
+            cpu_process_seconds_total,
+            memory_process_bytes,
+        }) = get_process_cpu_metric()
+        else {
+            panic!("unable to get current process's cpu metric");
+        };
```

Comment:
accepted. I will use the original `warn!` way although it is more stupid in my thinking.

Propagating is not an easier option otherwise the original code just do it. And the entire codebase has a lot of panics, I just follow that existing style and do not change any bad behaviors to the whole project. I personally think you have double standards. 😺

Reply:
The code in the diff I see still seems to be using `panic!`.

> the entire codebase has a lot of panics

The rule we follow (or try to) is to not panic in response to conditions at runtime.
The use of panicking macros and methods like `Result::except` is permitted if the developer believes that the failure case cannot occur at runtime (for example, when making assertions about invariants).
In other words, a panic at runtime should always indicate a bug.
Conditions that can occur in a correctly functioning application should be handled without panics.
`Result` is a common way to achieve that.

Reply:
@weekday-grandine-io 

> The use of panicking macros and methods like Result::except is permitted if the developer believes that the failure case cannot occur at runtime 

Your point of view is actually specious. Bugs or poorly thought-out code can invalidate invariants. Without a review of the entire system, who can determine whether a branching is for the invariant or just as an ordinary condition?

The best practice is to completely disable panic unless you want fail-fast.

For these codes, I actually forgot to change this code. If throwing errors here, a lot of code needs to be changed. I have a headache here. The original logic is to use `unwrap_or_default`. This implementation is really not good because you can't distinguish whether it is an error or a true zero value.

If you are OK, I want to use follow the idea of `unwrap_or_default`. That's if error, we just use the default value (zero here). This makes life easier.  I will change this in the weekend.